ext/standard/php_uuid.h

```diff
@@ -0,0 +1,163 @@
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | http://www.php.net/license/3_01.txt                                  |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Richard Fussenegger <php@fleshgrinder.com>                   |
+  +----------------------------------------------------------------------+
+*/
+
+/* $Id$ */
+
+#ifndef PHP_UUID_H
+#define PHP_UUID_H
+
+#include "zend_types.h"
+#include "php.h"
+
+PHPAPI extern zend_class_entry *php_ce_UUID;
+PHPAPI extern zend_class_entry *php_ce_UUIDParseException;
+
+#define PHP_UUID_LEN 16
+typedef struct php_uuid {
+	uint8_t bytes[PHP_UUID_LEN];
+} php_uuid;
+
+#define PHP_UUID_HEX_LEN 33
+typedef struct php_uuid_hex {
+	char str[PHP_UUID_HEX_LEN];
+} php_uuid_hex;
+
+#define PHP_UUID_STRING_LEN 37
+typedef struct php_uuid_string {
+	char str[PHP_UUID_STRING_LEN];
+} php_uuid_string;
+
+typedef enum php_uuid_variant {
+	PHP_UUID_VARIANT_NCS             = 0,
+	PHP_UUID_VARIANT_RFC4122         = 1,
+	PHP_UUID_VARIANT_MICROSOFT       = 2,
+	PHP_UUID_VARIANT_FUTURE_RESERVED = 3,
+} php_uuid_variant;
+
+static const uint8_t PHP_UUID_VERSION_1_TIME_BASED      = 1;
+static const uint8_t PHP_UUID_VERSION_2_DCE_SECURITY    = 2;
+static const uint8_t PHP_UUID_VERSION_3_NAME_BASED_MD5  = 3;
+static const uint8_t PHP_UUID_VERSION_4_RANDOM          = 4;
+static const uint8_t PHP_UUID_VERSION_5_NAME_BASED_SHA1 = 5;
+
+BEGIN_EXTERN_C()
+
+PHPAPI void php_uuid_create_v3(php_uuid *uuid, const php_uuid *namespace, const char *name, const size_t name_len);
+
+PHPAPI int php_uuid_create_v4(php_uuid *uuid, const zend_bool throw);
+
+static zend_always_inline int php_uuid_create_v4_silent(php_uuid *uuid)
+{
+	return php_uuid_create_v4(uuid, 0);
+}
+
+static zend_always_inline int php_uuid_create_v4_throw(php_uuid *uuid)
+{
+	return php_uuid_create_v4(uuid, 1);
+}
+
+PHPAPI void php_uuid_create_v5(php_uuid *uuid, const php_uuid *namespace, const char *name, const size_t name_len);
+
+static const zend_always_inline php_uuid_variant php_uuid_get_variant(const php_uuid *uuid)
+{
+	if ((uuid->bytes[8] & 0xC0) == 0x80) return PHP_UUID_VARIANT_RFC4122;
+	if ((uuid->bytes[8] & 0xE0) == 0xC0) return PHP_UUID_VARIANT_MICROSOFT;
+	if ((uuid->bytes[8] & 0x80) == 0x00) return PHP_UUID_VARIANT_NCS;
+	return PHP_UUID_VARIANT_FUTURE_RESERVED;
+}
+
+static const zend_always_inline uint8_t php_uuid_get_version(const php_uuid *uuid)
+{
+	return uuid->bytes[6] >> 4;
+}
+
+static const zend_always_inline php_uuid php_uuid_namespace_dns()
+{
+	return (php_uuid) { "\x6b\xa7\xb8\x10\x9d\xad\x11\xd1\x80\xb4\x00\xc0\x4f\xd4\x30\xc8" };
```

Comment:
This syntax is not C89 compatible.

Reply:
What became of the C99 discussion again?

Reply:
Guess it petered out, once again... It might be time to just PR the switch to C99 and see if anyone complains.

Reply:
I'm in. 😉

Reply:
maybe for PHP 8.0? :p

but seriously we should allow C99 asap